lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,393 @@
+        type_arg = '-g'
+        quota_default = 'root'
+        if name is None:
+            name = quota_default
+
+        if 'gquota' not in mp['mntopts'] and 'grpquota' not in mp['mntopts'] and 'gqnoenforce' not in mp['mntopts']:
+            module.fail_json(
+                msg='%s is not mounted with the gquota/grpquota/gqnoenforce option. (current options: %s)'
+                    % (mountpoint, mp['mntopts']), **result
+            )
+        try:
+            grp.getgrnam(name)
+        except KeyError as e:
+            module.fail_json(msg='User %s doesn\'t exist.' % name, **result)
+
+    elif quota_type == 'project':
+        type_arg = '-p'
+        quota_default = '#0'
+        if name is None:
+            name = quota_default
+
+        if 'pquota' not in mp['mntopts'] and 'prjquota' not in mp['mntopts'] and 'pqnoenforce' not in mp['mntopts']:
+            module.fail_json(msg='%s is not mounted with the pquota/prjquota/pqnoenforce option.' % mountpoint, **result)
+
+        if name != quota_default and not os.path.isfile('/etc/projects'):
+            module.fail_json(msg='/etc/projects doesn\'t exist.', **result)
+
+        if name != quota_default and not os.path.isfile('/etc/projid'):
+            module.fail_json(msg='/etc/projid doesn\'t exist.', **result)
+
+        if name != quota_default and name is not None and get_project_id(name) is None:
+            module.fail_json(msg='%s hasn\'t been defined in /etc/projid.' % name, **result)
+
+        prj_set = True
+        if name != quota_default:
+            cmd = 'project %s' % name
+            r = exec_quota(module, cmd, mountpoint)
+            if r['rc'] != 0:
+                result['cmd'] = cmd
+                result['retvar'] = r
+                module.fail_json(msg='Could not get project state.', **result)
+            else:
+                for line in r['stdout']:
+                    if '%s - project identifier is not set' in line:
+                        prj_set = False
+                        break
+
+        if not prj_set and not module.check_mode:
+            cmd = 'project -s'
+            r = exec_quota(module, cmd, mountpoint)
+            if r['rc'] != 0:
+                result['cmd'] = cmd
+                result['retvar'] = r
+                module.fail_json(msg='Could not get quota realtime block report.', **result)
+            else:
+                result['changed'] = True
+        elif not prj_set and module.check_mode:
+            result['changed'] = True
+
+    # Set limits
+    if state == 'absent':
+        bhard = 0
+        bsoft = 0
+        ihard = 0
+        isoft = 0
+        rtbhard = 0
+        rtbsoft = 0
+
+    current_bsoft, current_bhard = quota_report(module, mountpoint, name, quota_type, 'b')
+    current_isoft, current_ihard = quota_report(module, mountpoint, name, quota_type, 'i')
+    current_rtbsoft, current_rtbhard = quota_report(module, mountpoint, name, quota_type, 'rtb')
+
+    result['xfs_quota'] = dict(
```

Comment:
No need to add everything in a separate dictionary.. It only makes it harder for users to get to the information.
As a user I would probably add:
```yaml
    register: xfs_quota
```
And with this code I need to access it as `xfs_quota.xfs_quota.bsoft`.
Keep it simple :-)

Reply:
So, stat is a bad example. Got it.

Reply:
resolved